pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -183,8 +184,12 @@ func (c *Controller) reconcileShootRequest(req reconcile.Request) (reconcile.Res
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	key := shootKey(shoot)
 
+	// if shoot is no longer managed by this gardenlet (e.g., due to migration to another seed) then don't requeue
+	if seedName := confighelper.SeedNameFromSeedConfig(c.config.SeedConfig); (len(seedName) > 0 && *shoot.Spec.SeedName != seedName) || (len(seedName) == 0 && !controllerutils.SeedLabelsMatch(c.seedLister, *shoot.Spec.SeedName, c.config.SeedSelector)) {
```

Comment:
This is the same code like in https://github.com/gardener/gardener/blob/aec56a2c408c6cd67e8fe7aef3a613c06559dafe/pkg/gardenlet/controller/shoot/shoot_care_control.go#L124, right? Can you extract it into a function that can be used on both places?